lib/src/pubspec.dart

```diff
@@ -105,6 +106,6 @@ class Pubspec {
     if (vc == null) {
       return false;
     }
-    return !vc.allows(_dart2First);
+    return !vc.allows(_dart2First) && !vc.allows(_dart2Last);
```

Comment:
Why don't we turn this into what sortie suggested and just ask: `!vc.allows(_dart2Last)`
If I understand it correctly that means: "warn if latest dart 2.x release is not supported", if we do this we should probably also change the name of the property...

Reply:
Oh, I was thinking about a bit more complex solution (passing in the version for both the check and the error message). I guess using only _dart2Last with a message that is vague on the actual version works for us, I'll update this PR.